src/WebJobs.Extensions.NotificationHubs/Bindings/NotificationHubAsyncCollector.cs

```diff
@@ -4,23 +4,44 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.NotificationHubs;
+using Microsoft.Azure.WebJobs.Host;
 
 namespace Microsoft.Azure.WebJobs.Extensions.NotificationHubs
 {
     internal class NotificationHubAsyncCollector : IAsyncCollector<Notification>
     {
         private INotificationHubClientService _notificationHubclientService;
         private string _tagExpression;
+        private bool _enableTestSend;
+        private TraceWriter _traceWriter;
 
-        public NotificationHubAsyncCollector(INotificationHubClientService clientService, string tagExpression)
+        public NotificationHubAsyncCollector(INotificationHubClientService clientService, string tagExpression, bool enableTestSend, TraceWriter traceWriter)
         {
             _notificationHubclientService = clientService;
             _tagExpression = tagExpression;
+            _enableTestSend = enableTestSend;
+            _traceWriter = traceWriter;
         }
 
         public async Task AddAsync(Notification item, CancellationToken cancellationToken = default(CancellationToken))
         {
-            await _notificationHubclientService.SendNotificationAsync(item, _tagExpression);
+            NotificationOutcome notificationOutcome = await _notificationHubclientService.SendNotificationAsync(item, _tagExpression);
+            if (_enableTestSend)
+            {
+                _traceWriter.Info($"[NotificationHubsDebugLog] TrackingId = {notificationOutcome.TrackingId}");
+                if (notificationOutcome.Results == null || notificationOutcome.Results.Count == 0)
+                {
+                    _traceWriter.Info($"[NotificationHubsDebugLog] NotificationOutcome.Results is null or empty." +
+                                     "See https://azure.microsoft.com/en-us/documentation/articles/notification-hubs-push-notification-fixer/ for deatils on how to diagnose issues while sending push notifications");
+                }
+                else
+                {
+                    foreach (RegistrationResult result in notificationOutcome.Results)
+                    {
+                        _traceWriter.Info($"[NotificationHubsDebugLog] ApplicationPlatform:{result.ApplicationPlatform} RegistrationId:{result.RegistrationId} Outcome:{result.Outcome}");
```

Comment:
That's what I had assumed -- so using .Info() made sense since you're explicitly toggling it on/off for testing. Otherwise you'd have to EnableTestSend and then also change your logging to Verbose in order to see any results.